firmware/version_data.py

```diff
@@ -0,0 +1,99 @@
+#!/usr/bin/env python
+"""
+This file embeds in the firmware information like the git commit
+number,git branch information and git status of the build tree.
+"""
+
+import subprocess
+import os
+import tempfile
+import filecmp
+import shutil
+
+commit = subprocess.check_output(['git', 'log', '--format="%H"', '-n', '1'])\
+                                    .decode('utf-8')
+branch = subprocess.check_output(['git', 'symbolic-ref', '--short', 'HEAD'])\
+                                    .decode('utf-8')
+describe = subprocess.check_output(['git', 'describe', '--dirty'])\
+                                    .decode('utf-8')
+status = subprocess.check_output(['git', 'status', '--short']).decode('utf-8')
+length = status.count('\n')
+
+if "PLATFORM" in os.environ:
+    platform = os.environ['PLATFORM']
+else:
+    platform = ""
+    print("PLATFORM NOT SET")
+
+if "TARGET" in os.environ:
+    target = os.environ['TARGET']
+else:
+    target = ""
+    print("TARGET NOT SET")
+
+temp_h = tempfile.NamedTemporaryFile(suffix='.h', delete=True, mode='w+t')
```

Comment:
A small possible optimization here. As we are now in Python rather then shell, we probably don't need to use an actually on disk file here (IE A `tempfile.NamedTemporaryFile` object!). Instead we could just use a `StringIO` object, which is basically a mutable string that provides a file like interface. See example below;

Creating a file object using open();
```python
f = open("myfile.txt", "w+t", encoding="utf-8")
f.write("some text data")
f.seek(0)
assert f.read() == "some text data"
```

Using a StringIO object for an in-memory objects which looks like a file;
```python
f = io.StringIO()
f.write("some text data")
assert f.getvalue() == "some text data"
```